[Tests] Occasional failure: IndexLookupIT.testCallWithDifferentFlagsFails

Spate of occasional failures recently on master with this error:

	  1> [2017-02-16T05:53:41,605][INFO ][o.e.s.IndexLookupIT      ] [IndexLookupIT#testCallWithDifferentFlagsFails]: cleaned up after test
	  1> [2017-02-16T05:53:41,605][INFO ][o.e.s.IndexLookupIT      ] [testCallWithDifferentFlagsFails]: after test
	  2> REPRODUCE WITH: gradle :core:integTest -Dtests.seed=CB09BE6BB3CAF2BA -Dtests.class=org.elasticsearch.script.IndexLookupIT -Dtests.method="testCallWithDifferentFlagsFails" -Dtests.security.manager=true -Dtests.jvm.argline="-XX:-UseConcMarkSweepGC -XX:+UseG1GC" -Dtests.locale=fi-FI -Dtests.timezone=America/Miquelon
	FAILURE 0.36s J1 | IndexLookupIT.testCallWithDifferentFlagsFails <<< FAILURES!
	   > Throwable #1: java.lang.AssertionError: 
	   > Expected: <1>
	   >      but: was <2>
	   > 	at __randomizedtesting.SeedInfo.seed([CB09BE6BB3CAF2BA:DF6CD06AF0DAFC69]:0)
	   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	   > 	at org.elasticsearch.script.IndexLookupIT.testCallWithDifferentFlagsFails(IndexLookupIT.java:590)
	   > 	at java.lang.Thread.run(Thread.java:745)

Would not reproduce locally so opening this issue to track similar failures

--- a/core/src/test/java/org/elasticsearch/script/IndexLookupIT.java
+++ b/core/src/test/java/org/elasticsearch/script/IndexLookupIT.java
@@ -580,15 +580,19 @@ public class IndexLookupIT extends ESIntegTestCase {
                     .addScriptField("tvtest", script)
                     .get();
 
-            assertThat(numPrimaries, greaterThan(1));
+            // (partial) success when at least one shard succeeds
+            assertThat(numPrimaries, greaterThan(response.getShardFailures().length));
             assertThat(response.getFailedShards(), greaterThanOrEqualTo(1));
 
             for (ShardSearchFailure failure : response.getShardFailures()) {
                 assertThat(failure.reason(), containsString(expectedError));
             }
         } catch (SearchPhaseExecutionException e) {
-            assertThat(numPrimaries, equalTo(1));
-            assertThat(e.toString(), containsString(expectedError));
+            // Exception thrown when *all* shards fail
+            assertThat(numPrimaries, equalTo(e.shardFailures().length));
+            for (ShardSearchFailure failure : e.shardFailures()) {
+                assertThat(failure.reason(), containsString(expectedError));
+            }
         }
 
         // Should not throw an exception this way round

